manifest/manifest.go

```diff
@@ -266,3 +268,45 @@ func layerInfosToStrings(infos []LayerInfo) []string {
 	}
 	return layers
 }
+
+// compressionMIMETypeSet describes a set of MIME type “variants” that represent differently-compressed
+// versions of “the same kind of content”.
+// The map key is the return value of compression.Algorithm.Name(), or "" to mean "uncompressed";
+// the map value is a MIME type, or "" to mean "recognized but unsupported".
```

Comment:
Non-blocking nit: I'd prefer to make the "uncompressed" and "recognized but unsupported case" more explicit and make them consts. I was initially confused that the same value (i.e., "") can have very different meanings for keys and values.

It's well documented here but would make reading code using it more explicit.

Reply:
That’s a good point. Would using the same `""` value but named (`mtsUncompressed`? `mtsUnsupportedMIMEType`?) work? “Reserving” a real name worries me about conflicts with actual MIME type values.

(/me longingly looks at Swift’s variant types and enum type name inference.)

Reply:
Yes, that's what I meant. Using an empty string is fine. The consts are meant to only improve readability.